src/main/java/randoop/main/CloneVisitor.java

```diff
@@ -0,0 +1,1487 @@
+ */
+
+package randoop.main;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.ImportDeclaration;
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.PackageDeclaration;
+import com.github.javaparser.ast.TypeParameter;
+import com.github.javaparser.ast.body.AnnotationDeclaration;
+import com.github.javaparser.ast.body.AnnotationMemberDeclaration;
+import com.github.javaparser.ast.body.BodyDeclaration;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.ConstructorDeclaration;
+import com.github.javaparser.ast.body.EmptyMemberDeclaration;
+import com.github.javaparser.ast.body.EmptyTypeDeclaration;
+import com.github.javaparser.ast.body.EnumConstantDeclaration;
+import com.github.javaparser.ast.body.EnumDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.InitializerDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+import com.github.javaparser.ast.body.MultiTypeParameter;
+import com.github.javaparser.ast.body.Parameter;
+import com.github.javaparser.ast.body.TypeDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.body.VariableDeclaratorId;
+import com.github.javaparser.ast.comments.BlockComment;
+import com.github.javaparser.ast.comments.Comment;
+import com.github.javaparser.ast.comments.JavadocComment;
+import com.github.javaparser.ast.comments.LineComment;
+import com.github.javaparser.ast.expr.*;
+import com.github.javaparser.ast.stmt.*;
+import com.github.javaparser.ast.type.*;
+import com.github.javaparser.ast.visitor.GenericVisitor;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * CloneVisitor is a visitor which makes a deep copy of a JavaParser AST node. Unlike the
+ * CloneVisitor provided by JavaParser 2.4.0, this class also copies each node's orphan comments.
+ */
+public class CloneVisitor implements GenericVisitor<Node, Object> {
+
+	@Override
+	public Node visit(CompilationUnit _n, Object _arg) {
+		PackageDeclaration package_ = cloneNodes(_n.getPackage(), _arg);
+		List<ImportDeclaration> imports = visit(_n.getImports(), _arg);
+		List<TypeDeclaration> types = visit(_n.getTypes(), _arg);
+
+		return new CompilationUnit(
+				_n.getBeginLine(), _n.getBeginColumn(), _n.getEndLine(), _n.getEndColumn(),
+				package_, imports, types
+		);
+	}
+
+	@Override
+	public Node visit(PackageDeclaration _n, Object _arg) {
+		List<AnnotationExpr> annotations = visit(_n.getAnnotations(), _arg);
+		NameExpr name = cloneNodes(_n.getName(), _arg);
+		Comment comment = cloneNodes(_n.getComment(), _arg);
+
+		PackageDeclaration r = new PackageDeclaration(
+				_n.getBeginLine(), _n.getBeginColumn(), _n.getEndLine(), _n.getEndColumn(),
+				annotations, name
+		);
+		r.setComment(comment);
+    for (Comment oc : _n.getOrphanComments()) {
```

Comment:
This same three lines appars many times in the file.  How about abstracting it out into a method, to reduce the size of the diffs against upstream?